### PR TITLE
Fully automate dev setup with Gitpod

### PR DESCRIPTION
**Pull request by justaugustus**
**Overview**:

This commit implements a fully-automated development setup using Gitpod.io, an
online IDE for GitHub and GitLab that enables Dev-Environments-As-Code.
This makes it easy for anyone to get a ready-to-code workspace for any branch,
issue or pull request almost instantly with a single click.

**What problem does it solve?**:

Let's enable people to contribute to Dex from their browser! https://www.gitpod.io/

Not to mention, it's for a great cause! https://twitter.com/gitpod/status/1329141750196277249?s=20

**Special notes for a reviewer**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![Gitpod ready-to-code](https://img.shields.io/badge/Gitpod-ready--to--code-blue?logo=gitpod)](https://gitpod.io/#https://github.com/dexidp/dex)
```
Comment (sagikazarmark) [inline README.md:1]: Can you please move it down below the rest of the badges? Please also add `?style=flat-square`. Thanks!

**Review by sagikazarmark — CHANGES_REQUESTED** [review]

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+tasks:
```
Comment (sagikazarmark) [inline .gitpod.yml:1]: Can you add this file to `.dockerignore`?

**Comment by justaugustus** [comment]
@sagikazarmark -- should be good now!

**Review by sagikazarmark — APPROVED** [review]
LGTM

**Pull request merged**